src/Integration/LocalServices/ProjectSystemHelper.cs

```diff
@@ -186,6 +185,25 @@ public void AddFileToProject(Project project, string fullFilePath, string itemTy
             }
         }
 
+        public void RemoveFileFromProject(Project project, string fileName)
```

Comment:
The name doesn't match the behavior. This not only removes the file from the project, but removes the project too if it was empty and its a solution folder.

I tend to think that adding a dedicated method to remove empty solution folders would be cleaner.